### PR TITLE
Startup

### PR DESCRIPTION
**Pull request by anshsrtv**

### CODE REVIEW

File: startup/models.py [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+	pic = models.ImageField(upload_to='static/uploads/startups')
+	contact = models.TextField(max_length=13, null=True, blank=True)
+	url = models.URLField(null=False, blank=False, default="https://ecell.nitrr.ac.in")
+	founder = models.CharField(max_length=256)
+	address = models.TextField()
+	flag = models.BooleanField(default=False)
+	details = models.TextField(default='details coming soon')
+	created_at = models.DateTimeField(auto_now_add=True, editable=False)
+	modified_at = models.DateTimeField(auto_now=True, editable=False)
+
```
Comment (ethicalrushi) [inline startup/models.py:16]: 1) Make image,address an optional field
2) Use [this](https://www.python.org/dev/peps/pep-0008/#class-names) naming convention for classes

File: startup/views.py [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+
+
+@api_view(['GET',])
+def get_startups(request, year):
+
+    res_message = ""
+    res_status = ""
+    res_data = []
+
+    startups = startup.objects.filter(year=year, verified=True)
```
Comment (ethicalrushi) [inline startup/views.py:19]: Test this function locally

File: startup/views.py [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+@permission_classes((IsAdminUser,))
+def generate_spreadsheet(request):
+    response = HttpResponse(content_type='text/csv')
+    response['Content-Disposition'] = 'attachment; filename="startups.csv"'
+
+    writer = csv.writer(response)
+    writer.writerow(['Name', 'Company', 'Contact', 'Email', 'Experience', 'Year',
+                         'Verified', 'Description', 'Profile Pic', 'Social Media'])
+
+    startups = startup.objects.all().values_list('name', 'company', 'contact', 'email', 'experience',
```
Comment (ethicalrushi) [inline startup/views.py:71]: Test this function locally

File: startup/views.py [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+    if request.ecelluser.user_type in ['GST','VLT','CAB']:
+        return Response({
+            "message":"Unauthorized to view this page"
+        }, status=HTTP_401_UNAUTHORIZED)
+
+    res_message = ""
+    res_data    = []
+
+    startup = startupSerializer(data=request.data)
+
```
Comment (ethicalrushi) [inline startup/views.py:47]: Use validation for unique constraint to find if the entry already exists. If the entry already exists give user a message stating the same.

**Review by ethicalrushi — CHANGES_REQUESTED** [review]
Also pull the master and make sure that all checks are passed for your pr

**Review by naveennvrgup — COMMENTED** [review]
ansh do check the working of api with postman before making a pr

**Comment by naveennvrgup** [comment]
closed because it not relevant.

**Pull request closed without merging**